addons/point_of_sale/migrations/13.0.1.0.1/end-migration.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2020 ForgeFlow <https://www.forgeflow.com>
+# Copyright 2020 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def create_pos_payment_methods(env):
+    env.cr.execute("""
+        SELECT DISTINCT journal_id
+        FROM pos_config_journal_rel
+    """)
+    journal_ids = [aj[0] for aj in env.cr.fetchall()]
+    if journal_ids:
+        vals_list = []
+        for journal in env['account.journal'].browse(journal_ids):
+            vals = {
+                'name': 'Cash',
+                'cash_journal_id': journal.id,
+                'is_cash_count': True,
+                'company_id': journal.company_id.id,
+                'receivable_account_id': journal.company_id.account_default_pos_receivable_account_id.id,
+            }
+            vals_list += [vals]
+        env['pos.payment.method'].create(vals_list)
+    openupgrade.logged_query(
+        env.cr, """
+        INSERT INTO pos_config_pos_payment_method_rel
+            (pos_config_id, pos_payment_method_id)
+        SELECT pos_config_id, ppm.id
+        FROM pos_config_journal_rel pcjr
+        JOIN account_journal aj ON pcjr.journal_id = aj.id
+        JOIN pos_payment_method ppm ON ppm.cash_journal_id = aj.id""")
+
+
+def create_pos_payments(env):
+    env.cr.execute("""
+        SELECT DISTINCT absl.id, ppm.id
+        FROM account_bank_statement abs
+        JOIN pos_session ps ON abs.pos_session_id = ps.id
+        JOIN account_bank_statement_line absl ON absl.statement_id = abs.id
+        LEFT JOIN account_journal aj ON abs.journal_id = aj.id
+        JOIN pos_payment_method ppm ON ppm.cash_journal_id = aj.id
+    """)
+    st_lines_ids_dict = {stl[0]: stl[1] for stl in env.cr.fetchall()}
+    if st_lines_ids_dict:
+        vals_list = []
+        for st_line in env['account.bank.statement.line'].browse(
+                list(st_lines_ids_dict)):
+            vals = {
+                'name': st_line.name,
+                'pos_order_id': st_line.pos_statement_id.id,
+                'amount': st_line.amount,
+                'payment_method_id': st_lines_ids_dict[st_line.id],
+                'payment_date': st_line.create_date,
+            }
+            vals_list += [vals]
+        env['pos.payment'].create(vals_list)
```

Comment:
I don't remember if it's needed for something special, but each approach has it cons and pros. Do you have real measures for your statement?

Reply:
Well, when the migration is about table that can be big, SQL should be prefered. I don't have measures for this case, because I didn't tried to migrate my db with > 2,2 millions of records, but yes, introducing SQL has significant benefice, regarding duration. exemple : 
- https://github.com/OCA/OpenUpgrade/pull/2208 (from 5 hours to 1 hours)
- https://github.com/OCA/OpenUpgrade/pull/2151 (from 4 hours to 30 minutes)
- https://github.com/OCA/OpenUpgrade/pull/2148 (from 18 hours to 2 seconds)
- https://github.com/OCA/OpenUpgrade/pull/1934 (from hours to seconds)
- etc.

Reply:
Yes, but unless you have measures proving that this is significantly slow, we shouldn't do any effort of going to SQL, that needs to handle by hand everything.

Reply:
> but unless you have measures proving that this is significantly slow

I don't understand you on that point. It seems to me quite obvious that it will be _very_ slower. Let's think backwards. Are there cases where creating millions of records through the ORM takes less or as much time as an SQL query ?

Reply:
> My fear is that doing by SQL, there were things not computed and this can affect

That's another point, that I can understand.
Well on that current topic, this new model is quite simple, and I really don't fear side effect, even If I don't use 13.0 for the time being (See : https://github.com/odoo/odoo/blob/13.0/addons/point_of_sale/models/pos_payment.py)

But well, if it's not possible to do it in that PR, let's go ahead. I'll improve it if someone hasn't already done so, when I'll work on the migration to 16.

Reply:
Yeah, seen like that, it doesn't seem to be very complicated, but as said, I haven't touched it. Maybe when doing my review, I decide to convert it for the same reasons as you want. Are you serious about not doing the jump till v16?